docs/user-guide/data-structures/creating-arrays-and-datasets.ipynb

```diff
@@ -0,0 +1,527 @@
+   "source": [
+    "### Special DTypes\n",
+    "\n",
+    "Scipp has a number of `dtypes` that require some conversion when creating variables.\n",
+    "Notably [scipp.datetimes](../../generated/functions/scipp.datetimes.rst), [scipp.vectors](../../generated/functions/scipp.vectors.rst), and their scalar counterparts [scipp.datetime](../../generated/functions/scipp.datetime.rst), [scipp.vector](../../generated/functions/scipp.vector.rst).\n",
+    "As well as types for spatial transformations in [scipp.spatial](../../generated/modules/scipp.spatial.rst).\n",
+    "While variables of all of these dtypes can be constructed using `scipp.array` and `scipp.scalar`, the specialized functions offer more convenience and document their intent better.\n",
+    "\n",
+    "`scipp.datetimes` constructs an array of date-time-points.\n",
+    "It can be called either with strings in ISO 8601 format:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.datetimes(dims=['t'], values=['2021-01-10T01:23:45',\n",
+    "                                 '2021-01-11T23:45:01'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Or with integers which encode the number of time units elapsed since the Unix epoc.\n",
+    "See also [scipp.epoch](../../generated/functions/scipp.epoch.rst)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.datetimes(dims=['t'], values=[0, 1610288175], unit='s')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the unit is mandatory in the second case.\n",
+    "\n",
+    "`scipp.vectors` creates an array of 3-vectors.\n",
+    "It does so by converting a sequence or array with a length of 3 in its inner dimension:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.vectors(dims=['x'], values=[[1, 2, 3], [4, 5, 6]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Data Arrays\n",
+    "\n",
+    "There is essentially only one way to construct [data arrays](../../generated/classes/scipp.DataArray.rst), namely its initializer:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "x = sc.linspace('x', start=1.5, stop=3.0, num=4, unit='m')\n",
+    "a = sc.scalar('an attribute')\n",
+    "m = sc.array(dims=['x'], values=[True, False, True, False])\n",
+    "data = x ** 2\n",
+    "sc.DataArray(data, coords={'x': x}, attrs={'a': a}, masks={'m': m})"
```

Comment:
Should we mention that variables are not copied here? Possibly link to https://scipp.github.io/reference/ownership-mechanism-and-readonly-flags.html#Data-arrays?